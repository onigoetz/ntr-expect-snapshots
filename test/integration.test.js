import { describe, test } from "node:test";
import fs from "node:fs/promises";
import path from "node:path";

import initializeSnapshot from "../src/index.js";
import { expect } from "expect";

initializeSnapshot(import.meta.url);

const snapshotFile = path.resolve(
  path.dirname(import.meta.url.replace("file://", "")),
  "snapshots/integration.test.js.snap"
);

test("a snapshot", async () => {
  expect("Lorem ipsum dolor sit amet,").toMatchSnapshot();
  expect("consectetur adipiscing elit,").toMatchSnapshot();

  const snapshotContent = await fs.readFile(snapshotFile, "utf-8");
  expect(snapshotContent).toContain("Lorem ipsum dolor sit");
});

describe("parent", () => {
  test("nested", async () => {
    expect("Sub-snapshot").toMatchSnapshot();
  
    const snapshotContent = await fs.readFile(snapshotFile, "utf-8");
    expect(snapshotContent).toContain("Sub-snapshot");
  });
});
