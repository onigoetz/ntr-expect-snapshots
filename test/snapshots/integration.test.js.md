# Snapshot report for `test/integration.test.js`

The actual snapshot is saved in `integration.test.js.snap`.

Generated by [AVA](https://avajs.dev).

## a snapshot

> Snapshot 1

    'Lorem ipsum dolor sit amet,'

> Snapshot 2

    'consectetur adipiscing elit,'
