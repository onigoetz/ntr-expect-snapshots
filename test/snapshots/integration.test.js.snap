// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a snapshot//0`] = `"Lorem ipsum dolor sit amet,"`;

exports[`a snapshot//1`] = `"consectetur adipiscing elit,"`;

exports[`nested//0`] = `"Sub-snapshot"`;
